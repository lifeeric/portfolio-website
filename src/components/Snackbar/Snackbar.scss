@import "../Theme/variables";

.snackbar {
  position: fixed;
  bottom: 40px;
  left: 25px;
  padding: 12px 16px;
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),
    0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12);
  color: $whiteColor;
  border-radius: 4px;
  z-index: 1110;
  background: #4caf50;
  font-weight: 500;
  transition: 0.5s ease;
  transform: translateY(200%);

  // opening
  &.open {
    transform: translateY(0);
  }

  // close
  &__close {
    font-size: 18px;
    vertical-align: middle;
    padding-left: 10px;
    line-height: 1px;
    transition: 0.5s ease;

    // hover
    &:hover {
      cursor: pointer;
      color: $secondary;
    }
  }
}
