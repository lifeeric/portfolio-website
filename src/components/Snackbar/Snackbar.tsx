import * as React from "react"
import { IoMdCloseCircleOutline } from "react-icons/io"
import { FcApproval } from "react-icons/fc"

import "./Snackbar.scss"

interface Props {
  onClick: () => void
  message: string
  isOpen: boolean
}

export const Snackbar: React.FC<Props> = ({ onClick, message, isOpen }) => {
  return (
    <div className={`snackbar ${isOpen && "open"}`}>
      {message}
      <span className="snackbar__close" onClick={onClick}>
        <IoMdCloseCircleOutline />
      </span>
    </div>
  )
}
