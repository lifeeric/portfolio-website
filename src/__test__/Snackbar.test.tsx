import * as React from "react"

import { shallow } from "enzyme"

import { Snackbar } from "../components/Snackbar/Snackbar"

describe("<Snackbar />", () => {
  it("should render", () => {
    const wrapper = shallow(<Snackbar />)
    expect(wrapper).toMatchSnapshot()
  })
})
