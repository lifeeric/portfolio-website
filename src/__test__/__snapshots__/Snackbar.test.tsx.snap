// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> should render 1`] = `
<div
  className="snackbar undefined"
>
  <span
    className="snackbar__close"
  >
    <IoMdCloseCircleOutline />
  </span>
</div>
`;
